lib/AST/ASTVerifier.cpp

```diff
@@ -3086,9 +3086,10 @@ class Verifier : public ASTWalker {
       }
 
       // If a decl has the Throws bit set, the ThrowsLoc should be valid,
-      // and vice versa, unless the decl was imported, de-serialized, or
-      // implicit.
+      // and vice versa, unless the decl was imported, de-serialized,
+      // distributed (meaning the throws is implicit), or implicit.
       if (!AFD->isImplicit() &&
+          !AFD->isDistributed() &&
```

Comment:
This seems fine for now, but makes me think we should rethink the connection between the bit and the location. It gives 4 options in the space of states for the two values, and it seems now at least 3 of them are meaningful. Perhaps in the future we should wrap this complexity behind a method returning an enum to describe these states.

Reply:
<Discussed more and did a rewrite of this>